File: src/main/java/ca/yyx/hu/ui/BaseFragment.java
package ca.yyx.hu.ui;

import android.app.Fragment;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;

/**
 * @author algavris
 * @date 05/11/2016.
 */

public class BaseFragment extends Fragment {

    public Context getContext() {
        return getActivity();
    }

    public Intent registerReceiver(BroadcastReceiver receiver, IntentFilter filter) {
        return getActivity().registerReceiver(receiver, filter);
    }

    public void unregisterReceiver(BroadcastReceiver receiver) {
        getActivity().unregisterReceiver(receiver);
    }
}

File: src/main/java/ca/yyx/hu/app/BaseFragment.kt
package ca.yyx.hu.app

import android.app.Fragment
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter

/**
 * @author algavris
 * *
 * @date 05/11/2016.
 */

open class BaseFragment : Fragment() {

    fun registerReceiver(receiver: BroadcastReceiver, filter: IntentFilter): Intent? {
        return activity.registerReceiver(receiver, filter)
    }

    fun unregisterReceiver(receiver: BroadcastReceiver) {
        activity.unregisterReceiver(receiver)
    }
}
